### ISSUE TITLE
Problems with the Fetch API

### ISSUE THREAD

**Issue by pschaeffer** [microsoft/vscode#189761]
Type: <b>Bug</b>

I am not sure if this is really a problem with VS Code or not. I use the Fetch API in JavaScript. When I run the Fetch directly in JavaScript with a Chrome browser, I have no problems. The Fetch API executes fine. A very important point is that I don't see an OPTIONS request being sent. However, when I run the same Fetch using exactly the same JavaScript (using VS Code) I do see an OPTIONS being sent. Even worse, the Fetch (under JavaScript, under Chrome, under VS Code) does not wait for the OPTIONS to complete. The Fetch just fails with a 'failed to fetch' error. I don't understand this. Why should VS Code make any difference? What am I doing wrong? What additional information should I provide? How should this problem be debugged?

VS Code version: Code 1.81.0 (6445d93c81ebe42c4cbd7a60712e0b17d9463e97, 2023-08-02T12:37:13.485Z)
OS version: Windows_NT x64 10.0.19045
Modes:

<details>
<summary>System Info</summary>

|Item|Value|
|---|---|
|CPUs|Intel(R) Core(TM) i7-4702HQ CPU @ 2.20GHz (8 x 2195)|
|GPU Status|2d_canvas: enabled<br>canvas_oop_rasterization: disabled_off<br>direct_rendering_display_compositor: disabled_off_ok<br>gpu_compositing: enabled<br>multiple_raster_threads: enabled_on<br>opengl: enabled_on<br>rasterization: enabled<br>raw_draw: disabled_off_ok<br>video_decode: enabled<br>video_encode: unavailable_off<br>vulkan: disabled_off<br>webgl: enabled<br>webgl2: enabled<br>webgpu: enabled|
|Load (avg)|undefined|
|Memory (System)|15.91GB (5.79GB free)|
|Process Argv|--folder-uri file:///c%3A/Users/Peter/Documents/Visual%20Studio%20Code/Projects/WebApplication5/WebApplication5 --crash-reporter-id 7c17602d-d116-41bf-9eeb-6f288773aaef|
|Screen Reader|no|
|VM|0%|
</details><details><summary>Extensions (10)</summary>

Extension|Author (truncated)|Version
---|---|---
copilot|Git|1.100.306
python|ms-|2023.14.0
vscode-pylance|ms-|2023.8.10
jupyter|ms-|2023.7.1002162226
jupyter-keymap|ms-|1.1.2
jupyter-renderers|ms-|1.0.17
vscode-jupyter-cell-tags|ms-|0.1.8
vscode-jupyter-slideshow|ms-|0.1.5
live-server|ms-|0.4.9
LiveServer|rit|5.7.9

</details><details>
<summary>A/B Experiments</summary>

```
vsliv368cf:30146710
vsreu685:30147344
python383cf:30185419
vspor879:30202332
vspor708:30202333
vspor363:30204092
vstes627:30244334
vslsvsres303:30308271
vserr242:30382549
pythontb:30283811
vsjup518:30340749
pythonptprofiler:30281270
vshan820:30294714
vstes263:30335439
vscoreces:30445986
vscod805cf:30301675
binariesv615:30325510
bridge0708:30335490
bridge0723:30353136
vsaa593cf:30376535
pythonvs932:30410667
vsclangdf:30486550
c4g48928:30535728
dsvsc012cf:30540253
pynewext54:30695312
azure-dev_surveyone:30548225
vsccc:30803844
282f8724:30602487
f6dab269:30613381
showlangstatbar:30737416
vsctsb:30748421
03d35959:30757346
pythonfmttext:30731395
pythoncmvfstrcf:30756944
fixshowwlkth:30771522
showindicator:30805244
pythongtdpath:30769146
i26e3531:30792625
gsofb:30804716
pythonnosmt12:30797651
pythonidxpt:30805730
pythonnoceb:30805159
e537b577:30795824
dsvsc013:30795093
dsvsc014:30804076

```

</details>

**Comment by VSCodeTriageBot** [comment]
We closed this issue because it is a question about using VS Code rather than an issue or feature request. Please search for help on [StackOverflow](https://aka.ms/vscodestackoverflow), where the community has already answered thousands of similar questions. You may find their [guide on asking a new question](https://aka.ms/vscodestackoverflowquestion) helpful if your question has not already been asked. See also our [issue reporting guidelines](https://aka.ms/vscodeissuereporting).

Happy Coding!

**Comment by pschaeffer** [comment]
Hi,

 

This is not a question. This is (in my opinion) a bug/issue/problem/etc. Code (in JavaScript) that works properly in native mode fails under VSC. The problem is actually worse than it sounds. I was able to get rid of the OPTIONS HTTP request, by removing my headers from the request. However, I still get a ‘Failed to fetch’ error. 

 

Thank you

 

Peter Schaeffer

**Comment by gjsjohnmurray** [comment]
See #189774

**Comment by pschaeffer** [comment]
Hi,

 

Warning. This may be a duplicate.

 

I added what follows to my launch.json file and the problem went away.

 

"runtimeArgs": ["--disable-web-security","--user-date-dir=c:\\chrome-browser"]

 

Putting all these things together, it looks like when I ran without VSC, CORS is/was somehow (automatically? partially?) disabled. When I run inside VSC, CORS is not disabled. As a consequence, CORS must be manually disabled. At least this explanation fits the facts. What is really going on, I don’t know.

 

Thank you

 

Peter Schaeffer

 

P.S. I would suggest adding (in some form) this explanation to issue 189774, so other folks can search for it
